code/modules/mob/living/simple_animal/friendly/dog.dm

```diff
@@ -61,6 +71,73 @@
 			if(M && stat != DEAD) // Same check here, even though emote checks it as well (poor form to check it only in the help case)
 				emote("growl")
 
+
+/mob/living/simple_animal/pet/dog/handle_automated_movement()
+	. = ..()
+	handle_dogomated_movement()
+
+///This proc is used by dog subtypes to add on to their automated movements, so we don't get strange conflicts with ian down the line. Also lets ian be a good well trained perfect boy.
+/mob/living/simple_animal/pet/dog/proc/handle_dogomated_movement()
+	//Feeding, chasing food, FOOOOODDDD
+	if(IS_HORIZONTAL(src) || buckled)
+		return
+
+	if(++turns_since_scan > 5)
+		turns_since_scan = 0
+
+		// Has a target, but it's not where it was before, and it wasn't picked up by someone.
```

Comment:
This comment is not quite true, specifically "but it's not where it was before"

Reply:
Let me tweak it, one moment